src/repo.rs

```diff
@@ -1511,6 +1551,94 @@ impl Repo {
         }
     }
 
+    fn collect_sharepolicy_responses(&mut self) {
+        let mut decisions = Vec::new();
+        for repo_id in mem::take(&mut self.share_decisions_to_poll) {
```

Comment:
Here I would rather see `for (doc_id, repo_id)` process the share authorization of the document info, similar to how docstate is processed with storage and network futures.